src/Symfony/Component/Env/Dotenv.php

```diff
@@ -0,0 +1,380 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Env;
+
+use Symfony\Component\Env\Exception\FormatException;
+use Symfony\Component\Env\Exception\FormatExceptionContext;
+use Symfony\Component\Env\Exception\PathException;
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\ExceptionInterface as ProcessException;
+
+/**
+ * Manages .env files.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+final class Dotenv
+{
+    const VARNAME_REGEX = '[A-Z][A-Z0-9_]*';
+    const STATE_VARNAME = 0;
+    const STATE_VALUE = 1;
+
+    private $path;
+    private $cursor;
+    private $lineno;
+    private $data;
+    private $end;
+    private $state;
+    private $export;
+    private $values;
+
+    /**
+     * Loads one or several .env files.
+     *
+     * @param ...string A list of files to load
+     *
+     * @throws FormatException when a file has a syntax error
+     * @throws PathException   when a file does not exist or is not readable
+     */
+    public function load(/*...$paths*/)
```

Comment:
In php7 (not hhvm!) you can actually do `load(string ...$paths)`. In fact, this is just an array, just like `func_get_args()`